src/app/shared/components/course-form/course-form.component.html

```diff
@@ -19,38 +24,97 @@ <h3>Main Info</h3>
           placeholder="Input text"
           rows="4"
           cols="5"
+          formControlName="description"
         ></textarea>
-        <span id="descriptionErrorMessage" class="text-danger">Description is required.</span>
-      </div>
-      <div class="separator"></div>
-      <h3>Authors</h3>
-      <div class="form__control">
-        <div class="course__flex">
-          <!--Use the name `author` for the form control.-->
-          <input type="text" id="author" placeholder="Input Author Name"/>
-          <app-button id="createAuthor"></app-button>
-        </div>
-        <span id="authorErrorMessage">New author should contain only latin letters and numbers.</span>
-        <div class="course__authors">
-          <!--Use the name `authors` for the form array.-->
-          ::Authors::
-        </div>
+        <ng-container *ngIf="(submitted || description?.touched) && description?.hasError('required')">
+          <span id="descriptionErrorMessage" class="text-danger">Description is required.</span>
+        </ng-container>
+        <ng-container *ngIf="(submitted || description?.touched) && description?.hasError('minlength')">
+          <span id="descriptionErrorMessage" class="text-danger">Description must be minimum 2 characters long.</span>
+        </ng-container>
       </div>
       <div class="separator"></div>
       <h3>Duration</h3>
       <div class="form__control">
         <div class="course__flex">
-          <!--Use the name `duration` for the form array.-->
-          <input type="number" id="duration" placeholder="Input duration"/>
+          <input type="number" id="duration" placeholder="Minutes..." formControlName="duration"/>
           <div class="course__duration">
-            ::Duration::
+            {{ duration?.value ? (duration?.value | duration) : (0 | duration) }}
```

Comment:
You should move some logic to the pipe itself instead of writing some logic here. For example, if I enter some negative values, it will display something that is not about right:
![image](https://github.com/user-attachments/assets/a97149ad-5162-46cc-8c58-eef975e1f70e)

Reply:
nice catch , thank you!